src/distances.jl

```diff
@@ -42,3 +42,32 @@ Distance used when `v ∈ s` can be expressed as the intersection of level sets
 The distance is expressed as `norm([max(f_i(v), 0) ∀ i ∈ I], n)` with `p` a norm value accepted by `LinearAlgebra.norm`.
 """
 struct NormedEpigraphDistance{p} <: AbstractDistance end
+
+"""
+    projection_on_set(distance_definition, s, v, dual)
+
+Compute the projection of a vector `v` on a set `s`.
+
+Each set `S` implements at least `projection_on_set(d::DefaultDistance, v::T, s::S)`
+with `T` of appropriate type for members of the set.
+
+The keyword `dual` specifies whether to compute projection on the cone or its dual cone
```

Comment:
dual is not a keyword in that case but an argument

Reply:
removed dual now